### ISSUE TITLE
Advanced settings are ignored

### ISSUE THREAD

**Issue by leonbaronick** [yungnickyoung/Better-Caves#55]
I played around some more with the configuration, and found your wiki page about the advanced hidden settings. I tried to mess around with them, but they seem to be completely ignored. I generated 3 worlds with the same seed, but vastly different advanced settings, but the worlds are exactly the same. All other settings work fine. Just the advanced settings are ignored.

**Comment by yungnickyoung** [comment]
What version?

**Comment by leonbaronick** [comment]
1.12.2. Latest version of the mod for that version. And Better Caves is the only mod installed

**Comment by yungnickyoung** [comment]
Can you pastebin the debug log?

**Comment by leonbaronick** [comment]
Do I have to enable debug messages somewhere in the mod? I'm not exactly sure what "debug log" you're referring to. Or do you mean the normal log-file?

**Comment by yungnickyoung** [comment]
There should be a `debug.log` file. But actually I think I know the issue. Advanced params can only be applied in dimension-specific configs. This is something I totally failed to talk about in the wiki. Try applying the advanced settings in a dimension-specific config, e.g. `DIM0_config.cfg`, and see if it works. I'll update the wiki in the meantime.

**Comment by leonbaronick** [comment]
Tried it out, and it worked 👍 
Advanced settings work when they are dimension-specific. 

Is there a special reason why they don't exist on a global level?

**Comment by yungnickyoung** [comment]
Yeah, it's just a limitation of the architecture under the hood.

The base config uses Forge's config handling to read the file and save the options in variables, like any other normal Forge config. These values are used as the default for each dimension's config object (`ConfigHolder`).

Dimension-specific configs are read in by my own I/O handler and then applied to override any of the specified options for that dimension's `ConfigHolder`. So, configs using my handler can recognize the advanced settings, but the regular Forge config ignores the advanced settings so that they don't generate in the base config when the file is first created.

Basically, I decided that the inconvenience from not being able to use advanced settings in the base config was worth hiding those settings from users, since I didn't want a bunch of people reporting issues due to them mindlessly messing with those settings lol.

But yeah, my bad for the wiki being wrong. I've updated it to be as clear as possible so no one else will have to waste their time in the future.
